src/TestSuite/IntegrationTestTrait.php

```diff
@@ -281,7 +281,22 @@ public function enableRetainFlashMessages(): void
      */
     public function configRequest(array $data): void
     {
-        $this->_request = $data + $this->_request;
+        $this->_request = array_merge_recursive($data, $this->_request);
```

Comment:
Shouldn't this be backported to 4.x? `$this->_request = $data + $this->_request;` is clearly buggy because it overwrites the existing values.